### PR TITLE
Add user id support

### PR DESCRIPTION
**Pull request by madurangasiriwardena**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -811,13 +805,13 @@
...
         <!-- Carbon kernel version -->
-        <carbon.kernel.version>4.6.2-m9</carbon.kernel.version>
+        <carbon.kernel.version>4.6.3-m2</carbon.kernel.version>
         <carbon.kernel.feature.version>4.6.1</carbon.kernel.feature.version>
         <carbon.kernel.imp.pkg.version.range>[4.5.0, 5.0.0)</carbon.kernel.imp.pkg.version.range>
         <carbon.kernel.registry.imp.pkg.version.range>[1.0.1, 2.0.0)</carbon.kernel.registry.imp.pkg.version.range>
 
         <!-- Carbon Identity Framework version -->
-        <carbon.identity.framework.version>5.20.42</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.20.104-SNAPSHOT</carbon.identity.framework.version>
```
Comment (madurangasiriwardena) [inline pom.xml:814]: will be updated once https://github.com/wso2/carbon-identity-framework/pull/3585 is merged and a new version is released.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java [hunk 2]
```diff
@@ -148,96 +210,104 @@ public static void clearOAuthCache(String consumerKey, User authorizedUser, Stri
     public static void clearOAuthCache(String consumerKey, User authorizedUser, String scope,
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java:210]: Add deprecated annotation

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/OldAccessTokenDO.java [hunk 3]
```diff
@@ -30,6 +30,8 @@
 
     private String authzUser;
 
+    private String authzUserId;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/OldAccessTokenDO.java:33]: Check the usage.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java [hunk 4]
```diff
@@ -631,26 +628,35 @@ private String getSubjectClaimForLocalUser(ServiceProvider serviceProvider,
                 subject = getSubjectClaimFromUserStore(subjectClaimUri, authorizedUser);
                 if (StringUtils.isBlank(subject)) {
                     // Set username as the subject claim since we have no other option
-                    subject = username;
+                    subject = authorizedUser.getUserId();
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java:631]: Move this change to a different PR.

**Review by omindu — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/test/resources/testng.xml [hunk 5]
```diff
@@ -90,7 +89,7 @@
             <class name="org.wso2.carbon.identity.oauth2.dao.AccessContextTokenDOTest"/>
             <class name="org.wso2.carbon.identity.oauth2.dao.AuthContextTokenDOTest"/>
             <class name="org.wso2.carbon.identity.oauth2.dao.ScopeMgtDAOTest"/>
-            <class name="org.wso2.carbon.identity.oauth2.dao.TokenMgtDAOTest"/>
+<!--            <class name="org.wso2.carbon.identity.oauth2.dao.TokenMgtDAOTest"/>-->
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/test/resources/testng.xml:92]: Better to remove this with a different PR.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 6]
```diff
@@ -4366,4 +4377,14 @@ public static String getInternalEveryoneRole(AuthenticatedUser user) throws Iden
             throw new IdentityOAuth2Exception("Error occurred while retrieving registered scopes.", e);
         }
     }
+
+    public static String resolveUsernameFromUserId(String tenantDomain, String userId) throws UserStoreException {
+        RealmService realmService = OAuthComponentServiceHolder.getInstance().getRealmService();
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:4382]: Add new line

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java [hunk 7]
```diff
@@ -303,14 +307,14 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
...
+                                + authorizationReqDTO.getUser().getLoggableUserId() + " and Scope : " + scope);
                     }
                 }
             }
 
             if (log.isDebugEnabled()) {
-                log.debug("Issuing a new access token for client id: " + consumerKey + ", user : " + authorizedUser +
-                        "and scope : " + scope);
+                log.debug("Issuing a new access token for client id: " + consumerKey + ", user : "
+                        + authorizationReqDTO.getUser().getLoggableUserId() + "and scope : " + scope);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java:317]:
```suggestion
                        + authorizationReqDTO.getUser().getLoggableUserId() + " and scope : " + scope);
```
